CMakeLists.txt

```diff
@@ -68,8 +68,24 @@ endif()
 stlab_detect_thread_system(STLAB_DEFAULT_THREAD_SYSTEM)
 set(STLAB_THREAD_SYSTEM ${STLAB_DEFAULT_THREAD_SYSTEM} CACHE STRING "Thread system to use (win32|pthread|pthread-emscripten|pthread-apple|none)")
 
-option(STLAB_MINIMAL_TASK_POOL "Use the minimal task pool size (ON|OFF), implies `STLAB_TASK_SYSTEM=portable`" OFF)
-if(STLAB_MINIMAL_TASK_POOL)
+#
+# TODO (bmedina): When STLab v3.0 is released, remove the deprecated STLAB_MINIMAL_TASK_POOL option
+# and simplify the code below.
+#
+option(STLAB_MINIMAL_TASK_POOL "Deprecated: Use STLAB_TASK_POOL_SIZE=1." OFF)
+if (STLAB_MINIMAL_TASK_POOL)
+  message(WARNING "STLAB_MINIMAL_TASK_POOL is deprecated. Use STLAB_TASK_POOL_SIZE=1, instead.")
+  set(STLAB_TASK_POOL_SIZE 1)
+else()
+  set(STLAB_TASK_POOL_SIZE 0 CACHE STRING "Define a custom task pool size. Default of zero implies a pool size of std::thread::hardware_concurrency. Non-zero implies STLAB_TASK_SYSTEM=portable.")
+endif()
+
+# Ensure STLAB_TASK_POOL_SIZE is a non-negative number
+if(NOT STLAB_TASK_POOL_SIZE MATCHES "^[0-9]+$")
+  message(FATAL_ERROR "STLAB_TASK_POOL_SIZE must be a non-negative, decimal integer.")
+endif()
+
+if(STLAB_TASK_POOL_SIZE GREATER 0)
```

Comment:
Also need to check for 1 and set STLAB_MINIMAL_TASK_POOL accordingly, and error if both are set.